vllm/v1/sample/ops/topk_topp_sampler.py

```diff
@@ -138,8 +138,25 @@ def apply_top_k_top_p(
 
     This function sorts the logits tensor, which can be slow for large batches.
     """
-    if k is None and p is None:
+    if p is None:
+        if k is None:
```

Comment:
Do we have a unit test checking the correctness of this?

Reply:
We should really have blanket coverage for this kind of thing, including different combinations of parameters (i.e. top-k with/without top-p etc.). I'm not sure whether we do though. I will check and add a unit test to compare the two impls.